src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorRequestContextFactory.cs

```diff
@@ -6,23 +6,20 @@
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Razor.LanguageServer.EndpointContracts;
 using Microsoft.AspNetCore.Razor.LanguageServer.Extensions;
+using Microsoft.CodeAnalysis.Razor.Logging;
 using Microsoft.CommonLanguageServerProtocol.Framework;
+using Microsoft.Extensions.Logging;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer;
 
-internal class RazorRequestContextFactory : IRequestContextFactory<RazorRequestContext>
+internal class RazorRequestContextFactory(ILspServices lspServices) : IRequestContextFactory<RazorRequestContext>
 {
-    private readonly ILspServices _lspServices;
-
-    public RazorRequestContextFactory(ILspServices lspServices)
-    {
-        _lspServices = lspServices;
-    }
+    private readonly ILspServices _lspServices = lspServices;
 
     public Task<RazorRequestContext> CreateRequestContextAsync<TRequestParams>(IQueueItem<RazorRequestContext> queueItem, TRequestParams @params, CancellationToken cancellationToken)
     {
-        var logger = _lspServices.GetRequiredService<LoggerAdapter>();
+        var logger = _lspServices.GetRequiredService<IRazorLoggerFactory>().CreateLogger<RazorRequestContextFactory>();
```

Comment:
Can this logger instantiation be moved to the ctor instead?

Reply:
I don't believe so. There are a few services in CLaSP that are "base services" and end up being needed before the DI contained has finished being created, hence this taking an `ILspServices` and pulling from there.